app/assets/javascripts/miq_api.js

```diff
@@ -72,14 +72,24 @@
     });
   };
 
+  API.ws_destroy = function() {
+    document.cookie = 'ws_token=; path=/ws/notifications; Max-Age=0;'
+  };
+
   API.logout = function() {
     if (sessionStorage.miq_token) {
       API.delete('/api/auth');
     }
 
+    API.ws_destroy();
     delete sessionStorage.miq_token;
   };
 
+  API.init = function() {
+    API.autorenew();
+    miqInitNotifications();
```

Comment:
Feels wrong to add something quite unrelated to API code since there's no real direct dependency. 

Can you rewert the `autorenew` -> `init` change please and just call `miqInitNotifications` in `application.html.haml`?

Or move this init function outside the API code.